modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java

```diff
@@ -504,7 +504,7 @@ public ModelsMap postProcessModels(ModelsMap objs) {
             if (cm.isEnum && !cm.vendorExtensions.containsKey(this.zeroBasedEnumVendorExtension)) {
                 if (Boolean.TRUE.equals(this.zeroBasedEnums)) {
                     cm.vendorExtensions.put(this.zeroBasedEnumVendorExtension, true);
-                } else if (!Boolean.FALSE.equals(this.zeroBasedEnums)) {
+                } else {
```

Comment:
I don't think we want this change. When the value is false, then the user has opted out. It should only go in here if the value is null.

Reply:
thanks for the explanation. happy to revert the change.

Reply:
undo via 7df05a53b53